server/routes/handlers/helpers/viewWithRecallAndPerson.ts

```diff
@@ -26,6 +29,32 @@ async function getUserName(userId: string, token: string): Promise<string> {
   }
 }
 
+interface UserNames {
+  assessedByUserName?: string
+  bookedByUserName?: string
+  dossierCreatedByUserName?: string
+}
+
+const getUserNames = async (recall: RecallResponse, token: string): Promise<UserNames> => {
+  const { assessedByUserId, bookedByUserId, dossierCreatedByUserId } = recall
+  const [assessedByResult, bookedByResult, dossierCreatedByResult] = await Promise.allSettled([
+    assessedByUserId ? getUserName(assessedByUserId, token) : undefined,
```

Comment:
does this mean we make the call even if we havent set the userId?

Reply:
nope; if no userId, then undefined is passed to Promise.allSettled array and it resolves instantly

Reply:
oh yeh, I see that now. doh